symcrypt-sys/inc/precomp.h

```diff
@@ -0,0 +1,16 @@
+//
+// SymCrypt linux shared object library pre-compiled header file
+//
+// Copyright (c) Microsoft Corporation. Licensed under the MIT license.
+//
+
```

Comment:
You should not be duplicating the SymCrypt headers in your repo. They're subject to change and will likely get out-of-sync quickly as we continue development on SymCrypt.

Instead, you should either get them from a package on the host system (e.g. the SymCrypt package on Mariner includes the headers), or make SymCrypt a submodule of this project. The former is preferable, but might not be practical right now since we don't have SymCrypt packages for many common Linux distros (like Ubuntu).

Reply:
I agree, this is a workaround for prototyping.

> They're subject to change and will likely get out-of-sync quickly as we continue development on SymCrypt.

I do have a follow up questions;
Is SymCrypt backwards compatible with all updates? if that is the case then we can look at the idea of searching the system for the most up to date header files to include. 

At the moment, we are only taking a subsection of SymCryptAPIs, and we're doing some manual modifications to the binding files to ensure correctness ( DATA vs CODE export issue ). If we are already taking these precautions would it make sense to manually manage and update the SymCrypt bindings on a need to update basis? As an example; every major version SymCrypt update. My worry with getting the latest SymCrypt versions would cause breaking changes between versions.

> since we don't have SymCrypt packages for many common Linux distros (like Ubuntu).

Is there any plan in the future to include a pkg-config file to the SymCrypt repo for Linux users? From what I have seen this is the standard method for finding and linking libraries as well as linking in header files. Also what is the distribution story on Windows in the future? The plan is to have a SymCrypt.dll ship with Windows but where where would we get the corresponding header files?

Reply:
Resolving, we can table this and leave it as an ongoing discussion since it does not block the progression of other wrappers. Also made an issue to track.

Reply:
We use semantic versioning, meaning that we guarantee compatibility within a given major version. We're currently on major version 103. If there are breaking changes in the future, we will bump that to 104. Until then, future versions of 103.x will be compatible with existing versions.

If you add SymCrypt as a submodule of your repo, it will be pinned to the specific commit you choose. It won't change out from under you; you have to manually update the submodule to pull in new changes. I think this is a good solution until we have a more formal packaging system you can rely on (e.g. vcpkg or using the distro's package manager).

We do have a pkg-config file in the repo as well.

Please don't put this off for a future change. Duplicating files is asking for trouble; it allows the code to be forked and then it becomes difficult to undo that.